### ISSUE TITLE
More pkg-config name transforms (for libarchive, libcrypto, etc)

### ISSUE THREAD

**Issue by TheZeus121** [ziglang/zig#13551]
### Zig Version

0.11.0-dev.174+d823680e1

### Steps to Reproduce and Observed Behavior

Add `linkSystemLibrary("archive")` to `build.zig`.  Observe that Zig just adds `-larchive` without any `-L` or `-I` flags (it actually doesn't call `pkg-config --cflags --libs`) so it fails if libarchive is installed to a non-standard path.

Now replace that with `linkSystemLibrary("libarchive")` and pkg-config gets called properly but if you remove pkg-config, you get `-llibarchive` which doesn't work.

### Expected Behavior

For `linkSystemLibrary("archive")` to call `pkg-config --cflags --libs libarchive`.

`linkSystemLibraryPkgConfigOnly("libarchive")` is not a solution because if libarchive is installed in a standard location it should still be linkable without pkg-config, but currently there is no portable way to do this without checking if `pkg-config` is in PATH in `build.zig`.

---

Currently [`lib/std/build.zig`'s `runPkgConfig`](https://github.com/ziglang/zig/blob/a50ad04e8542d63262dabaede5fff52e41157d07/lib/std/build.zig#L2068) doesn't support the following "transform kinds" I've seen in the wild (on Homebrew and on Debian):
- -larchive -> pkg-config libarchive
- -lrsvg-2 -> pkg-config librsvg-2.0
- -lxml2 -> pkg-config libxml-2.0

Aside from libxml2, it seems it should be possible to go through the checks that are in place already just also check them for names with added "lib" prefix.

----

I'd tackle this issue myself but I don't feel like learning how to compile zig from source tonight 🙈

**Comment by matu3ba** [comment]
> I'd tackle this issue myself but I don't feel like learning how to compile zig from source tonight

You can use the instructions in https://github.com/ziglang/zig/wiki/Building-Zig-From-Source and on problems/unclear things, ask in a community place https://github.com/ziglang/zig/wiki/Community.

**Comment by BratishkaErik** [comment]
Related: https://github.com/ziglang/zig/issues/3671

**Comment by Jan200101** [comment]
this is implemented by https://github.com/ziglang/zig/pull/22069

**Comment by TheZeus121** [comment]
> this is implemented by [#22069](https://github.com/ziglang/zig/pull/22069)

You're right, it's been fixed forever and I didn't notice.  Thanks, I'll close the issue.
